### DISCUSSION TITLE
右手真ん中の1行のキーが反応しないです。

### DISCUSSION THREAD

**Discussion by d6ihbtzg** [Yowkees/keyball#490] [Q&A]
また14と20、16と22が同時押しされます。
原因教えていただけると助かり
ます。

![307572347-71fb56ee-c0e0-47c1-a7ac-692d965df4de](https://github.com/Yowkees/keyball/assets/2555027/60a43468-eed5-44cc-bb55-5a4b7c348ba2)

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

キーが一列で効かないという症状はProMicro周辺のはんだ付けが原因の可能性があります。添付しました写真の赤い丸で示した部分のはんだが上手く端子に馴染んでいないように見えますので、ひとまずこちらのはんだ付けをやり直してみてください。
![307627378-60a43468-eed5-44cc-bb55-5a4b7c348ba2](https://github.com/Yowkees/keyball/assets/143698349/412f0a03-e99f-4eed-ad26-a7ca2750fc8a)

同時押しされる症状についてはダイオードの向きが間違っている物がないか一度確認していただけますと幸いです。

**Reply by d6ihbtzg** [reply]
はんだつけ直して見ましたが、だめでした。
ダイオードの向きも再度
確認しましたが、問題ありませんでした。
![20240226_133348](https://github.com/Yowkees/keyball/assets/2555027/ed55252a-02b5-406a-bbb8-aa6a67198516)

他にやれることありますでしょうか？

**Reply by Ryo624** [reply]
ご対応いただきありがとうございます。

ProMicroの表側のはんだ付けも確認させていただいてもよろしいでしょうか。
また、左右のProMicroを入れ替えたとき症状がどうなるかをご確認いただけますと幸いです。

**Reply by d6ihbtzg** [reply]
表側になります
![1708925917264710126002684795295](https://github.com/Yowkees/keyball/assets/2555027/dd06bb58-5fef-4f4c-af70-63d2bd4c36b5)

左右入れ替えてみましたが、変わりなかったです。

**Reply by Ryo624** [reply]
写真の添付ありがとうございます。

ProMicroを左右入れ替えても変わりないということで了解しました。
P
roMicroの表側のはんだ付けをやり直すとどうなりますでしょうか。少し丸まって端子に馴染んでいない箇所があるように見えますので、はんだの形が山型に端子に馴染むよう意識してはんだをやり直していただけますと幸いです。

**Reply by d6ihbtzg** [reply]
表面もはんだし直しましたが変わらずでした。
![1708927856158409702984914067717](https://github.com/Yowkees/keyball/assets/2555027/b12ecff5-d846-47cf-90f5-0df707af0b54)

**Reply by Ryo624** [reply]
ご対応ありがとうございます。

添付しました写真の赤い丸で示した部分のはんだの広がりを除去できますでしょうか。
16と書かれた箇所の端子にはんだが馴染んでいないように見えますので、そちらのご対応もよろしくお願いいたします。

![307681715-b12ecff5-d846-47cf-90f5-0df707af0b54](https://github.com/Yowkees/keyball/assets/143698349/ba86ca50-a1be-4597-bb0d-2e5f6c747296)

**Reply by d6ihbtzg** [reply]
![20240226_215218](https://github.com/Yowkees/keyball/assets/2555027/b957e9bc-2dbb-4791-9d98-de26672e61df)
ハンダし直しました。ただ、完全にきれいに取ることはできなかったです。
裏面も再度ハンダし直しましたがやっぱりだめでした。
![20240226_215328](https://github.com/Yowkees/keyball/assets/2555027/2c5f9e91-8c30-4c46-a405-836fede89fb3)

**Reply by Ryo624** [reply]
ご対応ありがとうございます。

角度の問題かもしれませんが、表側の赤で囲った部分のはんだがうまく馴染んでいなかったり浮いているように見えますがどうでしょうか。
![307804261-b957e9bc-2dbb-4791-9d98-de26672e61df](https://github.com/Yowkees/keyball/assets/143698349/91b2f898-be04-4c1f-830d-b153f73b7132)

**Reply by d6ihbtzg** [reply]
再度ハンダしましたが変わりありませんでした。
![17090008844297476669749232035539](https://github.com/Yowkees/keyball/assets/2555027/ac390b67-c269-4011-83a2-1feef9d22c98)

すみませんがご確認のほどよろしくお願いします。

**Comment by Ryo624** [comment]
ご対応いただきましてありがとうございます。

もう一点だけ気になる箇所があり、添付させていただいたProMicro裏側の写真の赤い丸で示した端子のはんだが隣の端子に触れて見えますがどうでしょうか。
何度も対応していただいて大変申し訳ありませんが、もし触れている場合ははんだが触れないように対処していただけますと幸いです。

![307804824-2c5f9e91-8c30-4c46-a405-836fede89fb3](https://github.com/Yowkees/keyball/assets/143698349/ff741dfe-eaff-4a4f-a5e9-37e659921fad)

**Reply by d6ihbtzg** [reply]
取り除いてみました。他に考えられることはありますでしょうか。
![1709011211732434217942177736722](https://github.com/Yowkees/keyball/assets/2555027/16dbfbce-4140-43ef-967b-4b1b24b89bef)
![17090112580016037203478429595369](https://github.com/Yowkees/keyball/assets/2555027/1104e7fe-1d8d-4b44-8429-ccc122bd3028)

**Reply by Ryo624** [reply]
添付しました写真の青で囲んだTRRSジャックのはんだが十分馴染んでいないように見えますのでこちらもやり直していただけますでしょうか。
![308046266-16dbfbce-4140-43ef-967b-4b1b24b89bef](https://github.com/Yowkees/keyball/assets/143698349/783c93e5-71ba-46b8-abb5-c31cf5ae4b13)

**Reply by d6ihbtzg** [reply]
ジャックの部分ハンダし直しました。
![17090220939997104994102605565977](https://github.com/Yowkees/keyball/assets/2555027/a41ec9a6-5639-4afb-a366-b4f260a52e13)
キーソケットとダイオードとpromicroが通信してるか確認する方法はありますでしょうか？
